plugins/newsdownloader.koplugin/feeds.xml

```diff
@@ -0,0 +1,8 @@
+<?xml version="1.0" encoding="UTF-8"?>
```

Comment:
> Lua files are specyfic for lua and XML files are more general/standard and used all over Internet and PC
> you don't have to be famirial with LUA to edit XML files
> so more people are able to edit XML over LUA files
> you don't have to be a lua developer to edit XML files

I agree with you XML has a much larger audience compared to Lua syntax.

However, please keep in mind that many of koreader's users do not have technical background, so they might not even know what XML is. In fact, XML is not very friendly for a human to read.

Compare:

```xml
<?xml version="1.0" encoding="UTF-8"?>
<feeds>
	<!-- List your feeds here: -->
	<feed limit="2">http://www.pcworld.com/index.rss</feed>
	<feed limit="3">http://rss.cnn.com/rss/edition_world.rss</feed>
	<!-- set limit to "0" means no download, "-1" no limit. --> 
	<feed limit="0">http://feeds.reuters.com/reuters/UKSportsNews?format=xml</feed>
</feeds>
```

With:

```lua
return {
    -- list your feeds here:
    { "http://www.pcworld.com/index.rss", limit = 2 },
    { "http://rss.cnn.com/rss/edition_world.rss", limit = 3 },
    -- set limit to "0" means no download, "-1" no limit.
    { "http://feeds.reuters.com/reuters/UKSportsNews?format=xml", limit = 0 },
}
```

Which one do you think is more readable by human, especially to those non-technical users?

And for the current users, they already have to know a little bit of Lua to edit the existing setting files, add a new config in XML will make it worse because now users have to learn both Lua and XML. In fact, I have plan to migrate all configurations from Lua to a proper config DSL like yaml or toml:

```yaml
# list your feeds here:
- url: "http://www.pcworld.com/index.rss"
  limit: 1
- url: "http://rss.cnn.com/rss/edition_world.rss"
  limit: 2
# set limit to "0" means no download, "-1" no limit.
- url: "http://feeds.reuters.com/reuters/UKSportsNews?format=xml"
  limit: 0
```

XML is not just hard to read by human, but also can't be parsed efficiently by machines. There are many serialization formats that are much more efficient than XML. The only benefit of XML I can think of is flexibility, but a properly designed data structure should be able to be represented in a simple DSL like json.


> on Android we wouldn't be able to edit newsConfig.lua (this is one of the most important for me as I'm running android) so we wouldn't be able to change default feeds (in current implementation), [on Android feed xml config is copied to writeable location)

I have merged newsConfig.lua into main.lua in my PR. See explanations in my commit message: https://github.com/mwoz123/koreader/pull/1/commits/8b4e9b9345692e2d383d6df52874278a069a47b9

You will be able to edit it too if the lua config is copied to the same writeable location.

> I don't know howto create a clean and meaningful list in LUA config file & don't have time to learn more LUA

I understand this, that's why I am offering the help to finish this part for you ;)

I am happy to discuss more with you on this if you still think XML config is a better choice.